### PR TITLE
At/#4206 wallet connect subtype meta

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**

### CODE REVIEW

File: packages/web/app/lib/web3/PersonalWeb3.ts [hunk 1]
```diff
@@ -2,14 +2,12 @@ import { EthereumAddressWithChecksum, EthereumNetworkId } from "@neufund/shared"
...
 import { TxData } from "web3";
 
-import { EWalletSubType, EWalletType, TWalletMetadata } from "../../modules/web3/types";
+import { TWalletMetadata } from "../../modules/web3/types";
 import { Web3Adapter } from "./Web3Adapter";
 
 export interface IPersonalWallet {
   readonly web3Adapter: Web3Adapter;
   readonly ethereumAddress: EthereumAddressWithChecksum;
-  readonly walletType: EWalletType;
```
Comment (Moejoe90) [inline packages/web/app/lib/web3/PersonalWeb3.ts:11]: why remove this?!

This should stay as part of the `IPersonalWallet` interface

File: packages/web/app/modules/web3/types.ts [hunk 2]
```diff
@@ -13,6 +15,7 @@ export enum EWalletSubType {
   METAMASK = "METAMASK",
   PARITY = "PARITY",
   GNOSIS = "GNOSIS",
+  NEUFUND = "NEUFUND",
```
Comment (Moejoe90) [inline packages/web/app/modules/web3/types.ts:18]: is it neufund really? 

I would say this should be `WALLET_CONNECT` 
and as a subtype you have `NEUFUND`

File: packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts [hunk 3]
```diff
@@ -31,21 +33,63 @@ export class WalletConnectSessionTransactionError extends WalletError {
...
+  supportsRemoteKyc: boolean;
+  supportsWalletMigration: boolean;
+};
+
+const generateWalletMetaFormPeerMeta = (peerMeta: TPeerMeta | null): TWcMeta => {
+  const walletMeta = {
+    walletSubType: EWalletSubType.UNKNOWN as
+      | EWalletSubType.UNKNOWN
+      | EWalletSubType.METAMASK
+      | EWalletSubType.NEUFUND,
```
Comment (Moejoe90) [inline packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts:53]: Why not create a type out of this your already used it in another place

File: packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts [hunk 4]
```diff
@@ -31,21 +33,63 @@ export class WalletConnectSessionTransactionError extends WalletError {
...
+  signerType: ESignerType;
+  sessionRequestTimeout: number;
+  signingTimeout: number;
+  supportsExplicitTimeouts: boolean;
+  supportSessionPings: boolean;
+  supportsRemoteKyc: boolean;
+  supportsWalletMigration: boolean;
+};
+
+const generateWalletMetaFormPeerMeta = (peerMeta: TPeerMeta | null): TWcMeta => {
```
Comment (Moejoe90) [inline packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts:48]: Looks like a good case for unit tests

File: packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts [hunk 5]
```diff
@@ -31,21 +33,63 @@ export class WalletConnectSessionTransactionError extends WalletError {
...
+    signerType: ESignerType.ETH_SIGN,
+    signingTimeout: WC_DEFAULT_SIGN_TIMEOUT,
+    sessionRequestTimeout: WC_DEFAULT_SESSION_REQUEST_TIMEOUT,
+    supportsExplicitTimeouts: false,
+    supportSessionPings: false,
+    supportsRemoteKyc: false,
+    supportsWalletMigration: false,
+  };
+
+  if (peerMeta !== null && peerMeta.name === "Metamask") {
```
Comment (Moejoe90) [inline packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts:64]: why are we using direct strings this should be an Enum from the the subtype of the wallet

File: packages/web/app/lib/web3/PersonalWeb3.ts (hunk 1 above)
Comment (AlexeyTuboltsev) [reply packages/web/app/lib/web3/PersonalWeb3.ts:11]: oh, don't know what I had on mind. That's definitely a mistake, thanks for catching it

File: packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts (hunk 5 above)
Comment (AlexeyTuboltsev) [reply packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts:64]: but that's where I assign the subtype of wallet. PeerMeta comes from wc peer and it's a bunch of strings. Or what do you mean?

File: packages/web/app/modules/web3/types.ts (hunk 2 above)
Comment (AlexeyTuboltsev) [reply packages/web/app/modules/web3/types.ts:18]: that's exactly how it is. Did you forgot to expand the folded code? :D

File: packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts (hunk 3 above)
Comment (AlexeyTuboltsev) [reply packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts:53]: done

File: packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts (hunk 4 above)
Comment (AlexeyTuboltsev) [reply packages/web/app/lib/web3/wallet-connect/WalletConnectWallet.ts:48]: good point. 
done

File: packages/web/app/modules/web3/types.ts (hunk 2 above)
Comment (Moejoe90) [reply packages/web/app/modules/web3/types.ts:18]: ah my bad. it seemed as if we had it as a main type

**Pull request merged**